File: 559-MaximumDepthOfNAryTree/Python/main.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Node
        :rtype: int
        """
        self.max = 0
        
        def helper(root, depth):
            if not root:
                return
            
            self.max = max(depth, self.max)
            
            for child in root.children:
                helper(child, depth+1)
        
        helper(root, 1)
        return self.max

File: 559-MaximumDepthOfNAryTree/C++/main.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
public:
    int maxi = 0;
    
    void helper(Node* root, int depth) {
        if (!root) return;
        
        maxi = max(maxi, depth);
        
        for (auto child: root->children) {
            helper(child, depth+1);
        }
    }
    
    int maxDepth(Node* root) {
        helper(root, 1);
        return maxi;
    }
};